crystal_diffusion/models/mtp.py

```diff
@@ -0,0 +1,315 @@
+        assert os.path.exists(self.mlp_command), "mlp command not found in mlip-3 build folder"
+        self.mlp_templates = os.path.join(mlip_path, "MTP_templates")
+        assert os.path.exists(self.mlp_templates), "MTP templates not found in mlip-3 folder"
+        self.fitted_mtp = None
+        self.elements = None
+
+    def to_lammps_format(self):
+        """Write the trained MTP in a LAMMPS compatible format."""
+        # TODO
+        # write params write the fitted mtp in a LAMMPS compatible format
+        # self.write_param(
+        #    fitted_mtp=fitted_mtp,
+        #    Abinitio=0,
+        #    Driver=1,
+        #    Write_cfgs=predict_file,
+        #    Database_filename=original_file,
+        #    **kwargs,
+        # )
+        pass
+
+    def evaluate(self,
+                 test_structures: List[Structure],
+                 test_energies: List[float],
+                 test_forces: List[List[float]],
+                 test_stresses: Optional[List[List[float]]] = None,
+                 ) -> Tuple[pd.DataFrame, pd.DataFrame]:
+        """Evaluate energies, forces, stresses and MaxVol gamma factor of structures with trained MTP.
+
+        Args:
+            test_structures: evaluation set of pymatgen Structure Objects.
+            test_energies: list of total energies of each structure to evaluation in test_structures list.
+            test_forces: list of calculated (m, 3) forces of each evaluation structure with m atoms in structures list.
+                m can be varied with each single structure case.
+            test_stresses (optional): list of calculated (6, ) virial stresses of each evaluation structure in
+                test_structures list. If None, do not evaluate on stresses. Default to None.
+
+        Returns:
+            dataframe with ground truth energies, forces
+            dataframe with predicted energies, forces, MaxVol gamma (nbh grades)
+        """
+        original_file = "original.cfgs"
+        predict_file = "predict.cfgs"
+        test_structures, test_forces, test_stresses = check_structures_forces_stresses(
+            test_structures, test_forces, test_stresses
+        )
+        predict_pool = pool_from(test_structures, test_energies, test_forces, test_stresses)
+
+        with ScratchDir("."):  # mlip needs a tmp_work_dir - we will manually copy relevant outputs elsewhere
+            # write the structures to evaluate in a mlp compatible format
+            original_file = self.write_cfg(original_file, cfg_pool=predict_pool)
+            df_orig = self.read_cfgs(original_file, nbh_grade=False)  # read original values as a DataFrame
+
+            # calculate_grade is the method to get the forces, energy & maxvol values
+            cmd = [self.mlp_command, "calculate_grade", self.fitted_mtp, original_file, predict_file]
+            predict_file += '.0'  # added by mlp...
+            with subprocess.Popen(cmd, stdout=subprocess.PIPE) as p:  # run mlp
```

Comment:
This block is duplicated between `train` and `evaluate`. I would refactor that as a "private" _method.

Reply:
refactored